alacritty/src/config/monitor.rs

```diff
@@ -154,4 +160,19 @@ impl ConfigMonitor {
             warn!("config monitor shutdown failed: {err:?}");
         }
     }
+
+    /// Check if the config monitor needs to be restarted.
+    ///
+    /// This checks the supplied list of files against the monitored files to determine if a
+    /// restart is necessary.
+    pub fn needs_restart(&self, files: &[PathBuf]) -> bool {
+        self.watched_hash != Self::hash_paths(files)
+    }
+
+    /// Generate the hash for a list of paths.
+    fn hash_paths(files: &[PathBuf]) -> u64 {
+        let mut hasher = DefaultHasher::new();
+        Hash::hash_slice(files, &mut hasher);
```

Comment:
isn't it more correct to use the regular `hash` because the order of imports may change but not the imports themselves, because `hash_slice` is not accounting for such case and in such case we shouldn't restart the config reload, because nothing was changed.

Reply:
That's a good point which I hadn't considered.